### PR TITLE
Fix double TE registration against RC actor

### PR DESCRIPTION
**Pull request by fdc-ntflx**
### Context

Looks like the TE registers itself twice with the resource cluster actor in rapid succession which could lead to race conditions on caching artifacts.

* The first registration happens during [onStart](https://github.com/Netflix/mantis/blob/058657080e1341667b0fa38fb04b95d8160d560d/mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/TaskExecutor.java#L188)
* The second one when ZK registers the new resource cluster (which in the case of first bootstrap of the TE is the same as the one registered during the first registration

In order to avoid double registration for the same RC the High avail service can simply verify whether the RC has actually changed before trying to reconnect to it.

Note: I'm not sure if there are edge cases I'm not considering. I can't think of anything at the moment.

### Checklist

- [ ] `./gradlew build` compiles code correctly
- [ ] Added new tests where applicable
- [ ] `./gradlew test` passes all tests
- [ ] Extended README or added javadocs where applicable

### CODE REVIEW

File: mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/HighAvailabilityServicesUtil.java [hunk 1]
```diff
@@ -182,9 +182,14 @@ public void register(ResourceLeaderChangeListener<ResourceClusterGateway> change
               .observeOn(scheduler)
               .subscribe(nextDescription -> {
                 log.info("nextDescription={}", nextDescription);
+
+                ResourceClusterGatewayClient newResourceClusterGatewayClient = new ResourceClusterGatewayClient(clusterID, nextDescription);
+                if (newResourceClusterGatewayClient.equals(currentResourceClusterGateway)) {
+                    return;
```
Comment (Andyz26) [inline mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/HighAvailabilityServicesUtil.java:188]: nit: add metrics on both cases.

**Review by Andyz26 — APPROVED** [review]

File: mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/HighAvailabilityServicesUtil.java [hunk 2]
```diff
@@ -182,9 +182,14 @@ public void register(ResourceLeaderChangeListener<ResourceClusterGateway> change
               .observeOn(scheduler)
               .subscribe(nextDescription -> {
                 log.info("nextDescription={}", nextDescription);
+
+                ResourceClusterGatewayClient newResourceClusterGatewayClient = new ResourceClusterGatewayClient(clusterID, nextDescription);
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/HighAvailabilityServicesUtil.java:186]: It would be more appropriate to compare the descriptions rather than the clients themselves in terms of semantics. I’m concerned that if the logic for clients being equal changes in the future, it could cause problems for this logic.

File: mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/resourcecluster/ResourceClusterGatewayClient.java [hunk 3]
```diff
@@ -119,4 +121,13 @@ private AsyncHttpClient buildCloseableHttpClient() {
...
   }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        ResourceClusterGatewayClient gatewayClient = (ResourceClusterGatewayClient) o;
+        return clusterID.equals(gatewayClient.clusterID) &&
+            masterDescription.equals(gatewayClient.masterDescription);
+    }
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/resourcecluster/ResourceClusterGatewayClient.java:132]: Do you still need this?
